ha/util/conf_store.py

```diff
@@ -28,17 +28,23 @@
 from ha.k8s_setup.const import _DELIM, GconfKeys
 from cortx.utils.log import Log
 from ha.core.error import ClusterCardinalityError
+from ha.const import CLUSTER_GLOBAL_INDEX
+from cortx.utils.const import CLUSTER_CONF
 
 
 class ConftStoreSearch:
     """
     Wrapper class for using the confstore search APIs
     """
+    _cluster_conf_loaded = False
 
     def __init__(self, conf_store_req: bool = True):
         """Init method"""
         if conf_store_req:
             self._confstore = ConfigManager.get_confstore()
+        if not ConftStoreSearch._cluster_conf_loaded:
```

Comment:
Why do we need this check ?

Reply:
One time loading purpose.

Reply:
@mariyappanp, yes you don't need this check as the ConfigManager._safe_load() do this all

Reply:
Done.